### DISCUSSION TITLE
Creating OpenSSL win32 static libraries with runtime library linker option as MD

### DISCUSSION THREAD

**Discussion by aniljadaun** [openssl/openssl#22668] [Q&A]
Hi team

We are looking to create static libraries, by default non-shared libraries are created with MT runtime linker option. Can you please suggest on this how do create these static libs with MD flag?

**Comment by levitte** [comment]
We don't, and it's not necessary because we also use `/Zl`, which omits setting a default CRTL in the static library.  Building the *application* that uses the library determines what CRTL is being used.

**Reply by aniljadaun** [reply]
Hi @levitte 

thanks for your response. I am running into an issue with openssl static libs for win32 static . I have a lib named crypto.lib which is built and linked with statically built openssl libs and that crypto.lib is statically linked to my SecurityModule.dll and this dll is loaded in to my App at runtime . Now issue i am running is whenever i launch my app SecurityModule.dll  is loaded correctly and loadcount for this is displayed as 1 but as soon as any crypto.lib related workflow is triggered loadcount of SecurityModule.dll is changed to 0xffffff which means that securitymodule is  statically linked but with my understand that should no be the case. if i start using import library along with delay loaded dll i don't see this issue anymore. i think this could have something to do with default build option MT used for building of static libs . 
can you please also suggest some option which we can use to reduce the size of libcrypto .

**Reply by aniljadaun** [reply]
@t8m @paulidale  Can you please suggest on this ? I am stuck on this ,any idea on how to resolve this ? Previously we used openssl 1.0.2 and it worked fine for us.
